frontend/src/Components/Login/Login.jsx

```diff
@@ -45,13 +49,22 @@ const Login = ({ setAccessToken }) => {
          } else if (err.response.status === 417) {
             setErrorMsg("Wrong password");
          } else {
-            setErrorMsg("Unknown Error");
+            setErrorMsg(err.response.data.messages);
+            console.log(err.response.status);
          }
          setLoading(false);
          setForgotPassword(true);
+         setSuccessfulLogin("");
       }
    };
 
+   useEffect(() => {
+      if (token) {
+         emailVerification(token, setSuccessfulLogin, setErrorMsg, setInfoMsg, setSearchParams, setResendEmail);
```

Comment:
Too many params. I would pass f.e. onSuccess, onError and define those two callbacks